### PR TITLE
Issue #883: upgrade minimal required jdk version to 11

### PR DESCRIPTION
**Pull request by pbludov**
Issue #883 

Some Guava code replaced with JDK alternatives.

### CODE REVIEW

File: sevntu-checks/pom.xml [hunk 1]
```diff
@@ -25,7 +25,7 @@
     <maven.jacoco.plugin.version>0.8.7</maven.jacoco.plugin.version>
     <maven.pmd.plugin.version>3.11.0</maven.pmd.plugin.version>
     <pmd.version>6.38.0</pmd.version>
-    <java.version>1.8</java.version>
+    <java.version>11</java.version>
```
Comment (rnveach) [inline sevntu-checks/pom.xml:28]: When this is ready for merge, can you provide an example of using the Checkstyle JDK11 All JAR with Sevntu's JDK11 JAR and show a custom checks in Sevntu will work from the CLI using JDK11. I want  to verify an integration test to show sevntu works and we are not fooling ourselves with anything in our testing area.

I think manually testing will be enough, but I am not against from building a CI item to show this.

**Comment by romani** [comment]
I think we need to bind this update to bump of plugins to 10.x version.

**Comment by rnveach** [comment]
@pbludov it seems to me we can continue with this if eclipse-cs' bump is in maven.

**Comment by rnveach** [comment]
Eclipse-cs now upgraded.
https://github.com/checkstyle/eclipse-cs/releases/tag/10.0.0

**Comment by romani** [comment]
We need to move all jobs from wercker to Travis.

File: sevntu-checks/pom.xml (hunk 1 above)
Comment (romani) [reply sevntu-checks/pom.xml:28]:
```
rivanov@p5510:/var/tmp$ cat Test.java 
public class Test
{
    private final                                                                                                     MyThing thing;

}
rivanov@p5510:/var/tmp$ cat config.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="LineLengthExtended"/>
    </module>
</module>
rivanov@p5510:/var/tmp$ java -classpath sevntu-checks-1.41.0.jar:checkstyle-10.2-all.jar com.puppycrawl.tools.checkstyle.Main -c config.xml Test.java 
Starting audit...
[ERROR] /var/tmp/Test.java:3: Line is longer than 80 characters. Current line length is 132. [LineLengthExtended]
Audit done.
Checkstyle ends with 1 errors.
```

sevntu version looks old, but it is how it working now (no snapshot, new version appears during release, but I took this jar from target folder)

**Review by romani — APPROVED** [review]
ok to merge

Wercker could be ignored, I am not sure why build is failing, but we need to move all jobs out of wercker.

Travis eclipsecs build is stuck in queue, I tested it on local.

**Pull request merged**